### ISSUE TITLE
[Feature] (Horizontal) hardware screen stretch to get right ratio [sf#76]

### ISSUE THREAD

**Issue by openMSX-import** [openMSX/openMSX#100]
**Reported by atarulum on 2004-02-19 16:19 UTC**
Hi.

openMSX doesn't support Horizontal Stretch of MSX 
screen.

Vertical pixel size is same as original MSX's / MESS / 
ParaMSX.

But Horizontal pixel size is small than original MSX's / 
MESS / ParaMSX.

I think openMSX doesn't support Horizontal Stretch.

So I request to support Horizontal Stretch.

Here is sample screenshot of MSX2 from MESS.

**Comment by openMSX-import** [comment]
**Commented by atarulum on 2004-02-19 16:19 UTC**
This is MESS's screenshot of MSX2
Attached file MSX2JMESS.PNG:

Attachment at http://damad.be/joost/openmsx/attach/p/openmsx/feature-requests/_discuss/thread/0714362f/e137/attachment/MSX2JMESS.PNG

**Comment by openMSX-import** [comment]
**Commented by atarulum on 2004-02-19 16:20 UTC**
This is PARAMSX's screenshot of MSX2
Attached file MSX2JPARA.PNG:

Attachment at http://damad.be/joost/openmsx/attach/p/openmsx/feature-requests/_discuss/thread/0714362f/ae8d/attachment/MSX2JPARA.PNG

**Comment by openMSX-import** [comment]
**Commented by atarulum on 2004-02-19 16:20 UTC**
Logged In: YES 
user_id=641511

Here is sample screenshot of MSX2 from PARAMSX.

**Comment by openMSX-import** [comment]
**Updated by atarulum on 2004-02-19 16:20 UTC**
- **assigned_to**: nobody --> mthuurne
- **priority**: 5 --> 6

**Comment by openMSX-import** [comment]
**Commented by atarulum on 2004-02-19 16:21 UTC**
This is openMSX's screenshot of MSX2
Attached file MSX2JOPEN.png:

Attachment at http://damad.be/joost/openmsx/attach/p/openmsx/feature-requests/_discuss/thread/0714362f/f531/attachment/MSX2JOPEN.png

**Comment by openMSX-import** [comment]
**Commented by atarulum on 2004-02-19 16:21 UTC**
Logged In: YES 
user_id=641511

Here is sample screenshot of MSX2 from openMSX.

**Comment by openMSX-import** [comment]
**Updated by atarulum on 2004-02-19 16:29 UTC**
- **priority**: 6 --> 7

**Comment by openMSX-import** [comment]
**Commented by atarulum on 2004-02-22 13:34 UTC**
Real MSX2's screenshot from FS-4500
Attached file Msx2.gif:

Attachment at http://damad.be/joost/openmsx/attach/p/openmsx/feature-requests/_discuss/thread/0714362f/6c15/attachment/Msx2.gif

**Comment by openMSX-import** [comment]
**Commented by atarulum on 2004-02-22 13:34 UTC**
Logged In: YES 
user_id=641511

This is Real MSX2's Screenshot from National MSX2 FS-4500.
(MSX2.GIF)
(640x480) SCREEN 1:COLOR 15,4,7

I captured this by TV Card (PCI)

**Comment by openMSX-import** [comment]
**Commented by atarulum on 2004-02-22 13:38 UTC**
This is Modified openMSX's screenshot of MSX2
Attached file Msx2-2.gif:

Attachment at http://damad.be/joost/openmsx/attach/p/openmsx/feature-requests/_discuss/thread/0714362f/9c6b/attachment/Msx2-2.gif

**Comment by openMSX-import** [comment]
**Commented by atarulum on 2004-02-22 13:38 UTC**
Logged In: YES 
user_id=641511

Here is MSX2-2.GIF from openMSX's (758 x 505)
This was expanded the resolution of horizontal area (646 --
&gt; 758)
This actual size is just same as real MSX's.

In my opinion, openMSX had better support MESS's screen
aspect (6:5) to match up the resolution of horizontal area.

**Comment by openMSX-import** [comment]
**Commented by atarulum on 2004-05-26 19:05 UTC**
Logged In: YES 
user_id=641511

It is not necessary to support this any more.

**Comment by openMSX-import** [comment]
**Updated by atarulum on 2004-05-26 19:05 UTC**
- **priority**: 7 --> 1
- **status**: open --> wont-fix

**Comment by openMSX-import** [comment]
**Commented by atarulum on 2004-05-26 19:06 UTC**
Logged In: YES 
user_id=641511

It is not necessary to support this any more.

**Comment by openMSX-import** [comment]
**Updated by manuelbi on 2004-06-09 21:13 UTC**
- **priority**: 1 --> 3

**Comment by openMSX-import** [comment]
**Commented by manuelbi on 2004-06-09 21:13 UTC**
Logged In: YES 
user_id=78178

Reopened, since it is valid.
Wouter said SDL doesn't support this, but maybe it's
possible with openGL. (Maarten said it is...)

**Comment by openMSX-import** [comment]
**Updated by manuelbi on 2004-06-09 21:14 UTC**
- **status**: wont-fix --> open

**Comment by openMSX-import** [comment]
**Updated by manuelbi on 2005-05-14 09:29 UTC**
- **summary**: Request : Hardware (Horizontal) Stretch of MSX screen --> (Horizontal) hardware screen stretch to get right ratio

**Comment by openMSX-import** [comment]
**Commented by grauw on 2005-05-14 11:54 UTC**
Logged In: YES 
user_id=798154

When implementing this, please also remember that not all PC
screens have the same aspect ratio, e.g. the 1280xsomething
resolution. Also think of the users with a widescreen
display (now common on laptops and macs).

~Grauw

**Comment by openMSX-import** [comment]
**Commented by manuelbi on 2006-06-29 18:26 UTC**
Logged In: YES 
user_id=78178

This has been implemented now in the SDLGL-PP renderer,
meaning you will need OpenGL 2.0 support to see it.

To activate:

set display_deform horizontal_stretch

Thanks to Wouter for implementing it.

**Comment by openMSX-import** [comment]
**Updated by manuelbi on 2006-06-29 18:26 UTC**
- **assigned_to**: mthuurne --> m9710797
- **status**: open --> closed

**Comment by openMSX-import** [comment]
**Commented by dvik on 2006-06-29 18:32 UTC**
Logged In: YES 
user_id=1021645

Can you add the feature to OpenGL 1.1 as well? I'm pretty
sure its not too hard, just modify the coordinates when
calling glTexCoord2f.

Btw, will this option be available in Catapult as well? I
prefer to have this on as default.

**Comment by openMSX-import** [comment]
**Commented by manuelbi on 2006-06-29 18:47 UTC**
Logged In: YES 
user_id=78178

It won't be available in Catapult, because the current
Catapult is kind of dead. But you can easily set it up in
the console as I wrote in the previous message and then save
it "save_settings". Then it's default. (But only in
SDLGL-PP, at least for now.)

**Comment by openMSX-import** [comment]
**Commented by m9710797 on 2006-06-29 19:14 UTC**
Logged In: YES 
user_id=356949

While it's certainly possible to implement horizontal
stretch in our SDLGL renderer (OpenGL-1.x) it's not as
simple as adjusting texture coordinates. Internally we don't
have a single texture that contains the complete MSX image.
Instead we have lots of smaller textures that represent
characters (for MSX charater mode) or lines (for MSX bitmap
modes).

In the SDLGL-FB&lt;xx&gt; renderers we do have one texture for the
complete image, but because performance of these renderers
is not that great we don't enabled these in release versions.

So to summarize: while it's certainly technically possible
to implement horizontal stretch for SDLGL, (at this point in
time) it's not worth the effort IMHO.
